pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -150,6 +163,58 @@ func (d *depsCollector) populatePurls(ctx context.Context, docChannel chan<- *pr
 	return nil
 }
 
+// retrieve version data, scorecards, etc only on for a single package
+func (d *depsCollector) retrieveMetadataWithoutDeps(ctx context.Context, p datasource.Source, docChannel chan<- *processor.Document) error {
+	logger := logging.FromContext(ctx)
+
+	purl := p.Value
+	packageInput, err := helpers.PurlToPkg(purl)
+	if err != nil {
+		logger.Infof("failed to parse purl to pkg: %s", purl)
+		return nil
+	}
+
+	// check if top level purl has already been queried
+	if _, ok := d.checkedPurls[purl]; ok {
+		logger.Infof("purl %s already queried", purl)
+		return nil
+	}
+
+	// if version is not specified, cannot obtain accurate information from deps.dev. Log as info and skip the purl.
+	if *packageInput.Version == "" {
+		logger.Infof("purl does not contain version, skipping deps.dev query: %s", purl)
+		return nil
+	}
+
+	component := &PackageComponent{}
+	component.CurrentPackage = packageInput
+
+	err = d.collectAdditionalMetadata(ctx, packageInput.Type, packageInput.Namespace, packageInput.Name, packageInput.Version, component)
+	if err != nil {
+		logger.Debugf("failed to get additional metadata for package: %s, err: %v", purl, err)
+		return nil
+	}
+	logger.Infof("obtained additional metadata for package: %s", purl)
+
+	d.checkedPurls[purl] = component
+	blob, err := json.Marshal(component)
+	if err != nil {
+		return err
```

Comment:
nit: `fmt.Errorf`

Reply:
I logged this error instead of returning it, in order to avoid creating a channel just for this marshaling error. Is that OK?